mosip-identity-plugin/src/main/java/io/mosip/esignet/plugin/mosipid/service/IdaAuthenticatorImpl.java

```diff
@@ -251,10 +207,93 @@ public List<KycSigningCertificateData> getAllKycSigningCertificates() throws Kyc
+     * @return {@link  KycAuthResult}
+     * @throws KycAuthException throws this exception when fail to get the response
+     */
+    private KycAuthResult doKycAuthentication(String relyingPartyId, String clientId, KycAuthDto kycAuthDto,
+                                              boolean claimsMetadataRequired) throws KycAuthException {
+        log.info("Started to build kyc-auth request with transactionId : {} && clientId : {}",
+                kycAuthDto.getTransactionId(), clientId);
+        try {
+            IdaKycAuthRequest idaKycAuthRequest = getIdaKycAuthRequest(kycAuthDto, claimsMetadataRequired);
+            helperService.setAuthRequest(kycAuthDto.getChallengeList(), idaKycAuthRequest);
+
+            //set signature header, body and invoke kyc auth endpoint
+            String requestBody = objectMapper.writeValueAsString(idaKycAuthRequest);
+            RequestEntity<String> requestEntity = RequestEntity
+                    .post(UriComponentsBuilder.fromUriString(claimsMetadataRequired ? kycAuthUrlV2 : kycAuthUrl)
+                            .pathSegment(relyingPartyId, clientId).build().toUri())
+                    .contentType(MediaType.APPLICATION_JSON_UTF8)
+                    .header(SIGNATURE_HEADER_NAME, helperService.getRequestSignature(requestBody))
+                    .header(AUTHORIZATION_HEADER_NAME, AUTHORIZATION_HEADER_NAME)
+                    .body(requestBody);
+            ResponseEntity<IdaResponseWrapper<IdaKycAuthResponse>> responseEntity = restTemplate.exchange(requestEntity,
+                    new ParameterizedTypeReference<>() {
+                    });
+
+            if(responseEntity.getStatusCode().is2xxSuccessful() && responseEntity.getBody() != null) {
+                IdaResponseWrapper<IdaKycAuthResponse> responseWrapper = responseEntity.getBody();
+                if(responseWrapper.getResponse() != null && responseWrapper.getResponse().isKycStatus() &&
+                        responseWrapper.getResponse().getKycToken() != null) {
+                    if(claimsMetadataRequired){
+                        return new KycAuthResult(responseEntity.getBody().getResponse().getKycToken(),
+                                responseEntity.getBody().getResponse().getAuthToken(),
+                                responseEntity.getBody().getResponse().getVerifiedClaims());
+                    }
+                    return new KycAuthResult(responseEntity.getBody().getResponse().getKycToken(),
+                            responseEntity.getBody().getResponse().getAuthToken());
+                }
+                assert responseWrapper.getResponse() != null;
+                log.error("Error response received from IDA KycStatus : {} && Errors: {}",
+                        responseWrapper.getResponse().isKycStatus(), responseWrapper.getErrors());
+                throw new KycAuthException(CollectionUtils.isEmpty(responseWrapper.getErrors()) ?

```

Comment:
@Md-Humair-KK  can you try avoiding nested if blocks?

Reply:
Now inside if is changed to ternary operator evaluation, outside ifs are unavoidable I guess because the KycAuthResult(responseEntity.getBody().getResponse().getKycToken(),
                            responseEntity.getBody().getResponse().getAuthToken()); is invoked checking those conditions and I need verifiedclaims as well along with this for kyc auth v2 flow. Hence added claimsmetadatarequired flag check.